### PR TITLE
multiStringAllPositions() with non-const needle

### PR DESCRIPTION
**Pull request by rschu1ze**
With this PR, function `multiStringAllPositions()` supports non-const needle arguments. It was the last one of the string search functions which lacked support for that.

Originally motivated by @Avogar in https://github.com/ClickHouse/ClickHouse/pull/38485#issuecomment-1176274651.

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
SQL function multiStringAllPositions() now accepts non-const needle arguments.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by rschu1ze** [comment]
Log for the test failures:

```
2022-07-13 20:46:56 00233_position_function_family:                                         [ FAIL ] 127.68 sec. - Test runs too long (> 60s). Make it faster.
```

Since I have only added a handful of quick tests, I think the slowdown was related to infrastructure problems.

File: src/Functions/FunctionsMultiStringPosition.h [hunk 1]
```diff
@@ -17,112 +17,86 @@
...
-  *
-  * multiSearchAny(haystack, [pattern_1, pattern_2, ..., pattern_n]) -- find any of the const patterns inside haystack and return 0 or 1
-  * multiSearchAnyUTF8(haystack, [pattern_1, pattern_2, ..., pattern_n])
-  * multiSearchAnyCaseInsensitive(haystack, [pattern_1, pattern_2, ..., pattern_n])
-  * multiSearchAnyCaseInsensitiveUTF8(haystack, [pattern_1, pattern_2, ..., pattern_n])
-
-  * multiSearchFirstIndex(haystack, [pattern_1, pattern_2, ..., pattern_n]) -- returns the first index of the matched string or zero if nothing was found
-  * multiSearchFirstIndexUTF8(haystack, [pattern_1, pattern_2, ..., pattern_n])
-  * multiSearchFirstIndexCaseInsensitive(haystack, [pattern_1, pattern_2, ..., pattern_n])
-  * multiSearchFirstIndexCaseInsensitiveUTF8(haystack, [pattern_1, pattern_2, ..., pattern_n])
```
Comment (Avogar) [inline src/Functions/FunctionsMultiStringPosition.h:38]: Why did you delete it?

**Review by Avogar — APPROVED** [review]

File: src/Functions/FunctionsMultiStringPosition.h (hunk 1 above)
Comment (rschu1ze) [reply src/Functions/FunctionsMultiStringPosition.h:38]: Some of the functions were already documented in other files, but after checking closely, some were not. Will fix.

**Comment by Avogar** [comment]
Oops, I accidentally pressed the button `Update brunch`, I wanted to merge it instead. Let's wait for minimal required checks and merge it
